metricbeat/module/haproxy/haproxy.go

```diff
@@ -251,18 +253,21 @@ func (p *unixProto) run(cmd string) (*bytes.Buffer, error) {
 
 	conn, err := net.Dial(p.Network, p.Address)
 	if err != nil {
-		return response, err
+		return response, errors.Wrap(err, "Error connecting")
 	}
 	defer conn.Close()
 
 	_, err = conn.Write([]byte(cmd + "\n"))
 	if err != nil {
-		return response, err
+		return response, errors.Wrap(err, "Error writing to connection")
 	}
 
-	_, err = io.Copy(response, conn)
+	recv, err := io.Copy(response, conn)
 	if err != nil {
-		return response, err
+		return response, errors.Wrap(err, "error reading response")
+	}
+	if recv == 0 {
```

Comment:
Should we add a similar check to `httpProto.Stat()`?

Reply:
The HTTP one doesn't really have an equivalent that I can see?

Reply:
Well, if the response body is empty [here](https://github.com/elastic/beats/pull/14211/commits/b3f7d54d21aab2abb360324a115916fb54d33b5c#diff-92c341eac8789d66da3758d2043ed6f1R319), but not sure when if this is as likely to happen as in the tcp case...